src/main/java/uk/ac/cam/cl/dtg/isaac/api/GameboardsFacade.java

```diff
@@ -323,6 +335,71 @@ public final Response getFastTrackConceptProgress(@Context final Request request
         }
     }
 
+    /**
+     * REST endpoint to retrieve every question (and its status) associated with a particular FastTrack gameboard and
+     * history.
+     *
+     * @param request usually used for caching.
+     * @param httpServletRequest so that we can extract the users session information if available.
+     * @param gameboardId the unique id of the FastTrack gameboard which links the questions.
+     * @param upper the latest upper level question that is in the history.
+     * @param currentConceptTitle the concept title that the user is currently working on.
+     * @return a Response containing a list of augmented gameboard items for the gamebaord-concept pair or an error.
+     */
+    @GET
+    @Path("fasttrack/{gameboard_id}/concepts_with_upper/")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "Get the progress of the current user at a FastTrack gameboard.")
+    public final Response getFastTrackConceptFromHistory(@Context final Request request,
+                                                         @Context final HttpServletRequest httpServletRequest,
+                                                         @PathParam("gameboard_id") final String gameboardId,
+                                                         @QueryParam("upper") final String upper,
+                                                         @QueryParam("conceptTitle") final String currentConceptTitle) {
+
+        try {
+            Map<String, List<String>> fieldsToMatch = Maps.newHashMap();
+            fieldsToMatch.put(TYPE_FIELDNAME, Arrays.asList(FAST_TRACK_QUESTION_TYPE));
+            fieldsToMatch.put(ID_FIELDNAME + "." + UNPROCESSED_SEARCH_FIELD_SUFFIX, Arrays.asList(upper));
+            ResultsWrapper<ContentDTO> resultsList = this.api.findMatchingContent(this.contentIndex,
+                    SegueContentFacade.generateDefaultFieldToMatch(fieldsToMatch), null, null);
+
+            String conceptTitle = "";
+            if (resultsList.getTotalResults() == 1) {
+                conceptTitle = resultsList.getResults().get(0).getTitle();
+            }
+
+            if (!fastTrackGamebaordIds.contains(gameboardId)) {
+                return new SegueErrorResponse(Status.NOT_FOUND, "Gameboard id not a valid FastTrack gameboard id.")
+                        .toResponse();
+            }
+            AbstractSegueUserDTO randomUser = this.userManager.getCurrentUser(httpServletRequest);
```

Comment:
My fault because it is copied form the method above but this is not a random user it is the current user.